lambda/src/main/scala/pricemigrationengine/migrations/Newspaper2025P1Migration.scala

```diff
@@ -75,6 +99,42 @@ object Newspaper2025P1Migration {
   // Helpers
   // ------------------------------------------------
 
+  def getLabelFromMigrationExtraAttributes(item: CohortItem): Option[String] = {
+    for {
+      attributes <- item.migrationExtraAttributes
+    } yield {
+      val data: Newspaper2025ExtendedAttributes =
+        upickle.default.read[Newspaper2025ExtendedAttributes](attributes)
+      data.brandTitle
+    }
+  }
+
+  // (Comment Group: 571dac68)
+
+  def getNotificationData(
+      cohortSpec: CohortSpec,
+      item: CohortItem
+  ): ZIO[Zuora, Failure, Newspaper2025P1NotificationData] = {
+    MigrationType(cohortSpec) match {
+      case Newspaper2025P1 => {
+        (for {
+          brandTitle <- getLabelFromMigrationExtraAttributes(item)
+        } yield Newspaper2025P1NotificationData(
+          brandTitle
+        )) match {
+          case None =>
+            ZIO.fail(
+              DataExtractionFailure(
+                s"Could not build Newspaper2025P1NotificationData for item ${item.toString}"
+              )
+            )
+          case Some(d) => ZIO.succeed(d)
+        }
+      }
+      case _ => ZIO.succeed(Newspaper2025P1NotificationData(""))
```

Comment:
should this ever happen? Is it an error?

Reply:
I see. Perhaps the intent would be a little clearer if this function returns `ZIO.succeed(None)` in this case - currently you eventually wrap the value in a Some (`newspaper2025_brand_title`) anyway.